spin/__main__.py

```diff
@@ -19,35 +20,53 @@
 
 
 def main():
-    if not os.path.exists("pyproject.toml"):
-        print("Error: cannot find [pyproject.toml]")
+    def error(message):
+        print(f"Error: {message}", file=sys.stderr)
         sys.exit(1)
 
-    with open("pyproject.toml", "rb") as f:
-        try:
-            toml_config = tomllib.load(f)
-        except tomllib.TOMLDecodeError:
-            print("Cannot parse [pyproject.toml]")
-            sys.exit(1)
-
-    project_config = toml_config.get("project", {})
-
-    try:
-        config = toml_config["tool"]["spin"]
-    except KeyError:
-        print("No configuration found in [pyproject.toml] for [tool.spin]")
-        sys.exit(1)
-
-    proj_name = project_config.get("name", config["package"])
+    def load_toml(filename):
+        if not os.path.exists(filename):
+            return None
+        with open(filename, "rb") as f:
+            try:
+                return tomllib.load(f)
+            except tomllib.TOMLDecodeError:
+                error("cannot parse [{filename}]")
+
+    toml_config = collections.ChainMap()
+    toml_config.maps.extend(
```

Comment:
Any reason not to instantiate `ChainMap` directly from these dicts? We could have `load_toml` above return `{}` instead of None.

Reply:
Well, I wanted to add the loaded configurations as children of the ChainMap, in case in the future we need to store things manually on it (that would go to the (currently empty) first dict in the chain (`maps` attribute)). Regarding the returning `None`, I wanted to preserve that return value as a signal that the file is not present. Anyway, the change `None` -> `{}` is inconsequential for the rest of the code, as an empty dict will not be appended to the chain.

PS: Just in case I was not clear enough, a ChainMap created with no args is born with an initial empty dict in the chain. Appending to the `maps` list adds new dicts to the chain.

Reply:
Makes sense; I'm not sure we'd ever need to distinguish between manual parameters and the rest, but at this point we're in the weeds, so in it goes.

Reply:
> I'm not sure we'd ever need to distinguish between manual parameters and the rest

Think about third parties writing their own commands wrapping other commands and wanting to mess around with config to hack/monkeypatch things.